cmake/app/boilerplate.cmake

```diff
@@ -247,7 +247,21 @@ else() # NOT FIRST_BOILERPLATE_EXECUTION
 
   # Have the child image select the same BOARD that was selected by
   # the parent.
+  # Unless parent was "ns" in which case we assume that the child images are
+  # all are secure, and should be build using the secure version of the board.
+  # It is assumed that only the root app will be built as non-secure.
+  # This is not a valid assumption as there might be multiple non-secure
+  # images defined. With this technique, it is not possible to have multiple
+  # images defined as non-secure.
+  # TODO: Allow multiple non-secure images by using Kconfig to set the
+  # secure/non-secure property rather than using a separate board definition.
   set(BOARD ${CACHED_BOARD})
+  if (${BOARD} MATCHES  ".*ns")
```

Comment:
Won't this break CI if we integration test Zephyr with qemu_xtensa ?

Reply:
does it end with "ns" ? I think it ends with "nsa"